rework the css for the tree of sub menu categories

flynnduism opened the pull request:
This PR updates the css for the left sidebar menu:

* reducing the font size to better accomodate the amount of text for the large index of contents
* improves styling to increase distinction between parent/child categories - fixes #13 
* improves styling `active` and `hover` states to try and make things clearer

## Screenshot:

<img width="721" alt="Screen Shot 2021-05-14 at 2 23 07 PM" src="https://user-images.githubusercontent.com/686194/118333210-2f95d080-b4c0-11eb-9c65-a759b7912fee.png">

---

## A Note on Directories:

This task was fairly complex, as the nesting of subfolders has gotten pretty deep in the docs tree of contents. In some cases things are nested three or even four levels deep. Two sections in particular were rendering weirdly (I think the underlying Docsy theme is unable to handle that many layers of nesting) so I had to change two pages to not be inside additional directories:

```
content/docs/design_concepts/osm_components_interactions/_index.md
content/docs/install/manual_demo/_index.md
```

changed to:

```
content/docs/design_concepts/osm_components_interactions.md
content/docs/install/manual_demo.md
```

---

Signed-off-by: flynnduism <dev@ronan.design>

bridgetkromhout commented:
Awesome! Some notes...

> Two sections in particular were rendering weirdly (I think the underlying Docsy theme is unable to handle that many layers of nesting) so I had to change two pages to not be inside additional directories:

This is actually great and I intend to do this to _everything_ possible. This style of naming only was necessary when we were also trying to make the links work on github also.

Three things remain:

1) some merge conflicts on this PR, so we can't merge it quite yet

2) is there any way we could have space or horizontal rules or something in the sidebar? Looking at this, it is hard to tell if "High-level Software Architecture Components & Descriptions" is one link or two. Hovering over doesn't make it obvious to me either, in terms of visual changes:

![image](https://user-images.githubusercontent.com/2104453/118336848-1e58be00-b4d8-11eb-870e-51668a581d0d.png)

3) could you maybe reclaim and use more space on the left sidebar if we didn't use this space on the right for a fairly unneeded right sidebar?

![image](https://user-images.githubusercontent.com/2104453/118337005-74c5fc80-b4d8-11eb-90a9-c4ea4379b03d.png)

flynnduism commented:
Thanks @bridgetkromhout 

1. Rebased
2. I've updated the styling with more differentiation, horizontal breaks to denote sections, arrows to denote further children and more dramatic hover states
3. I think that would be a good improvement - two columns would be a lot less squishy. by default, the underlying Docsy framework is laid out like this, but it's certainly doable to change it to a two column layout. I will issue a seperate PR for that as it's a broader layout change from the text styling here. 

Screenshots:

<img width="305" alt="Screen Shot 2021-05-14 at 7 20 05 PM" src="https://user-images.githubusercontent.com/686194/118345545-16a21500-b4ea-11eb-9490-b629522a7e73.png">

<img width="310" alt="Screen Shot 2021-05-14 at 6 03 06 PM" src="https://user-images.githubusercontent.com/686194/118345539-1144ca80-b4ea-11eb-970c-0d8dbac23c97.png">

_The current menu is humongous!_

bridgetkromhout reviewed (approved):
This is a good improvement. Thanks!
